### PR TITLE
Add 'eslint-plugin-jsdoc' to devDependencies.

### PR DESCRIPTION
**Pull request by JSAssassin**
addresses #30

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -2,7 +2,11 @@ module.exports = {
   env: {
     node: true
   },
-  extends: ['digitalbazaar'],
+  extends: [
+    'digitalbazaar',
+    'eslint-config-digitalbazaar',
```
Comment (aljones15) [inline .eslintrc.js:7]: I believe you can drop `eslint-config-digitalbazaar` and just use the shortname `digitalbazaar` here.

File: lib/index.js [hunk 2]
```diff
@@ -306,20 +315,26 @@ api.remove = async ({actor, account, type} = {}) => {
...
+ *   types that do not hash token challenges.
+ * @param {Array}[options.authenticatedMethods=[]] - A list of identifiers for
+ *   other methods methods that have already been authenticated (useful for
+ *   multifactor verification where one method depends on other methods to
+ *   first be used).
  *
- * @return a Promise that resolves to an object containing the account ID,
- *   the account `email` and token information if verified, `false` if not.
+ * @returns {Promise<object>} - A Promise that resolves to an object containing
+ *   the account ID,the account `email` and token information if verified,
```
Comment (aljones15) [inline lib/index.js:336]:
```suggestion
 *   the account ID, the account `email` and token information if verified,
```

Comment (aljones15) [reply lib/index.js:336]: @mattcollier I will leave it to you if there should be a comma before the `and` in this sentence.

File: lib/index.js [hunk 3]
```diff
@@ -528,10 +548,13 @@ api.getAuthenticationRequirements = async ({actor, account}) => {
...
  *
- * @param requiredAuthenticationMethods the required authentication methods.
- * @param authenticatedMethods the authenticated methods.
+ * @param {object} options - The options to use.
+ * @param {Array} options.requiredAuthenticationMethods - The required
+ *   authentication methods.
+ * @param {Array} options.authenticatedMethods - The authenticated methods.
  *
- * @return a Promise that resolves to `true` if met, `false` if not.
+ * @returns {Promise} - A Promise that resolves to `true` if met, `false` if
```
Comment (aljones15) [inline lib/index.js:556]:
```suggestion
 * @returns {Promise<boolean>} - A Promise that resolves to `true` if met, `false` if
```

Comment (aljones15) [reply lib/index.js:556]: @JSAssassin this suggestion might break the eslint line length rule.

File: .eslintrc.js (hunk 1 above)
Comment (aljones15) [inline .eslintrc.js:7]:
```suggestion
```

File: lib/clients.js [hunk 4]
```diff
@@ -52,16 +52,23 @@ const PERMISSIONS = bedrock.config.permission.permissions;
...
+ * @param {string} [options.email] - The email address for the account to add
+ *   a client for.
+ * @param {string} options.clientId - An identifier for the client
+ *   (e.g., a cookie value).
+ * @param {boolean} [options.authenticated] - Set to `true` to indicate the
+ *   client has been authenticated; this should happen through the verification
+ *   of an authentication token associated with the client.
+ * @param {boolean} [options.notify=true] - Set to `true` to notify the account
+ *   user,
+ *   `false` not to.
```
Comment (davidlehn) [inline lib/clients.js:69]:
```suggestion
 *   user, `false` not to.
```

**Comment by aljones15** [comment]
@JSAssassin @davidlehn my apologies I marked Lehn's comment resolved. It might not be.

File: lib/index.js [hunk 5]
```diff
@@ -306,20 +315,26 @@ api.remove = async ({actor, account, type} = {}) => {
...
+ * @param {Array}[options.authenticatedMethods=[]] - A list of identifiers for
+ *   other methods methods that have already been authenticated (useful for
+ *   multifactor verification where one method depends on other methods to
+ *   first be used).
  *
- * @return a Promise that resolves to an object containing the account ID,
- *   the account `email` and token information if verified, `false` if not.
+ * @returns {Promise<object>} - A Promise that resolves to an object containing
+ *   the account ID, the account `email` and token information if verified,
+ * `false` if not.
```
Comment (davidlehn) [inline lib/index.js:337]:
```suggestion
 *   `false` if not.
```

Comment (JSAssassin) [reply lib/index.js]: Fixed.

**Pull request merged**